packages/engine/src/input/systems/ClientInputSystem.test.ts

```diff
@@ -7,63 +7,64 @@ import { GamepadAxis } from '../enums/InputEnums'
 import { InputType } from '../enums/InputType'
 import { InputValue } from '../interfaces/InputValue'
 import { InputAlias } from '../types/InputAlias'
-import { enableInput } from './ClientInputSystem'
+import ClientInputSystem, { enableInput } from './ClientInputSystem'
 
 describe('clientInputSystem', () => {
+  let world
+
+  beforeEach(() => {
+    world = createWorld()
+    Engine.currentWorld = world
+  })
     
-    it('check if enable input works', () => {
-        enableInput({ keyboard: true, mouse: true })
+  it('check if enable input works', () => {
+      enableInput({ keyboard: true, mouse: true })
 
-        strictEqual(Engine.keyboardInputEnabled, true)
-        strictEqual(Engine.mouseInputEnabled, true)
-    })
+      strictEqual(Engine.keyboardInputEnabled, true)
+      strictEqual(Engine.mouseInputEnabled, true)
+  })
 
-    it('check the input life cycle', () => {
-        const GAMEPAD_STICK = GamepadAxis.Left
-        const stickPosition: NumericalType = [0, 0, 0]
-        
-        Engine.inputState.set(GAMEPAD_STICK, {
-            type: InputType.TWODIM,
-            value: stickPosition,
-            lifecycleState: Engine.inputState.has(GAMEPAD_STICK) ? LifecycleValue.Started : LifecycleValue.Changed
-          })
+  it('check the input life cycle', async () => {
+      const GAMEPAD_STICK = GamepadAxis.Left
+      const GAMEPAD_STICKR = GamepadAxis.Right
+      const stickPosition: NumericalType = [0, 0, 0]
+      
+      Engine.inputState.set(GAMEPAD_STICK, {
+          type: InputType.TWODIM,
+          value: stickPosition,
+          lifecycleState: LifecycleValue.Started
+        })
+
+      strictEqual(Engine.inputState.size, 1)
         
-          strictEqual(Engine.inputState.size, 1)
+	    const clientInputSystem = await ClientInputSystem(world)
+      clientInputSystem()
```

Comment:
we should split ClientInputSystem into a bunch of smaller functions and run those as unit tests